### DISCUSSION TITLE
the question of light modes

### DISCUSSION THREAD

**Discussion by SuoXX** [T-Troll/alienfx-tools#390] [Q&A]
Hello author, my M16R1 is using AlienFX, and it recognizes the keyboard light, tail light, logo light, and power light. However, during the setup, I found that only color and pulse modes can be used for the tail light and logo light, and the keyboard light can only be set to a color without any modes. How can this be resolved?

**Comment by T-Troll** [comment]
APIv5 (AKA internal notebook keyboard) doesn't have per-key hardware effects except global ones. But you can emulate it using Grid effects.
About morph - it will not work until you set something using sliders (speed/length) at Colors tab. And you need AT LEAST two Actions to make it work (it's hardware limitation as well).
